app/controllers/shopify_app/callback_controller.rb

```diff
@@ -69,7 +69,7 @@ def update_rails_cookie(api_session, cookie)
     end
 
     def redirect_to_app
-      if ShopifyAPI::Context.embedded?
+      if ShopifyAPI::Context.embedded? && !fully_formed_url?(session[:return_to])
```

Comment:
While I see what this is doing now because I have the context, it isn't super clear as to the intent of the feature. Maybe we add another `return_to` local override that says `return_to = return_url if full_formed_url?`. Something similar to line 74's `return_to` override.

Reply:
Makes sense, I've refactored a bit so it's hopefully more clear. Let me know what you think